### PR TITLE
fixed a bug in Flurry source each method

### PR DESCRIPTION
**Pull request by TamerShlash**

### CODE REVIEW

**Comment by omarkhan** [comment]
@TamerShlash did you manage to get a failing test?

**Comment by TamerShlash** [comment]
@omarkhan I tried to do everything to make the test fail but it didn't! I pushed this bugfix only to let @T-Dnzt use this branch, we shouldn't merge the PR before fixing the tests which seem to be very lazy and never fail.

**Comment by Anumodana** [comment]
:+1:

**Comment by T-Dnzt** [comment]
Here's a failing test. The problem: No yield from Flurry = no `expect` called in the `each` loop.

File: spec/optimus_prime/sources/flurry_spec.rb [hunk 1]
```diff
@@ -32,14 +32,16 @@
...
-      expect(event.keys).to include(
-        'Session',
-        'Version',
-        'Device',
-        'Event',
-        'Timestamp',
-      )
-    end
+    events = []
+    source.each { |event, list| events << event }
```
Comment (omarkhan) [inline spec/optimus_prime/sources/flurry_spec.rb]: What is the `list` parameter here?

Comment (omarkhan) [inline spec/optimus_prime/sources/flurry_spec.rb]: What about this:

```ruby
events = source.to_a
```

Comment (T-Dnzt) [inline spec/optimus_prime/sources/flurry_spec.rb]: Forgot to remove it from `each_with_object` :p

**Comment by TamerShlash** [comment]
:+1:

**Comment by T-Dnzt** [comment]
:+1:

**Pull request merged**